classes/form/reportfilter_form.php

```diff
@@ -33,27 +33,44 @@ public function definition() {
         global $DB;
 
         $mform = $this->_form;
-        $errortypes = $this->_customdata['errortypes'];
+        $reportid = $this->_customdata['reportid'];
         $eventnames = $this->_customdata['eventnames'];
-        $responses = $this->_customdata['responses'];
         $resend = $this->_customdata['resend'];
 
+        if ($reportid == XAPI_REPORT_ID_ERROR) {
+            $errortypes = $this->_customdata['errortypes'];
+            $responses = $this->_customdata['responses'];
+        } elseif ($reportid == XAPI_REPORT_ID_HISTORIC) {
+            $eventcontexts = $this->_customdata['eventcontexts'];
```

Comment:
selecting an `eventcontext` in the form doesn't filter the results
![image](https://user-images.githubusercontent.com/51955245/81838346-e83bd380-953d-11ea-8f5a-be84ee3143d6.png)

Reply:
I have fixed this if you wish to test

Reply:
sorted 👍